sqlmesh/core/loader.py

```diff
@@ -79,6 +80,18 @@ def load(self, context: Context) -> LoadedProject:
         self._path_mtimes.clear()
         self._dag = DAG()
 
+        config_mtimes: t.Dict[Path, t.List[float]] = defaultdict(list)
+        for context_path, config in self._context.configs.items():
+            for config_file in context_path.glob("config.*"):
+                self._track_file(config_file)
+                config_mtimes[context_path].append(self._path_mtimes[config_file])
```

Comment:
If you want, you could augment _track_file to take a type like self._track_file(config_file, FileType.Config) and then add helpers get_mtimes(FileType.Config). Not sure its worth it, but I'll continue to throw out ideas to reduce caching specific code from the loader and you can do what you want with them. :-)